xacro/substitution_args.py

```diff
@@ -0,0 +1,576 @@
+    """
+    if len(args) != 2:
+        raise SubstitutionException('$(find-resource pkg path) '
+                                    'command only accepts two argument [%s]' % a)
+    before, after = _split_command(resolved, a)
+    path = _sanitize_path(args[1])
+    # we try to find the specific path in share via catkin
+    # which will search in install/devel space and the source folder of the
+    # package
+    from catkin.find_in_workspaces import find_in_workspaces
+    paths = find_in_workspaces(
+        ['share'], project=args[0], path=path, first_matching_workspace_only=True,
+        first_match_only=True, source_path_to_packages=source_path_to_packages)
+    if not paths:
+        raise SubstitutionException(
+            '$(find-resource pkg path) could not find path [%s]' % a)
+    return before + paths[0] + after
+
+
+def _split_command(resolved, command_with_args):
+
+    cmd = '$(%s)' % command_with_args
+    idx1 = resolved.find(cmd)
+    idx2 = idx1 + len(cmd)
+    return resolved[0:idx1], resolved[idx2:]
+
+
+def _separate_first_path(value):
+
+    idx = value.find(' ')
+    if idx < 0:
+        path, rest = value, ''
+    else:
+        path, rest = value[0:idx], value[idx:]
+    return path, rest
+
+
+def _sanitize_path(path):
+
+    path = path.replace('/', os.sep)
+    path = path.replace('\\', os.sep)
+    return path
+
+
+def _get_executable_path(base_path, path):
+
+    full_path = os.path.join(base_path, path)
+    if os.path.isfile(full_path) and os.access(full_path, os.X_OK):
+        return full_path
+    return None
+
+
+def _eval_arg(name, args):
+
+    try:
+        return args[name]
+    except KeyError:
+        raise ArgException(name)
+
+
+def _arg(resolved, a, args, context):
+    """
+    Process $(arg) arg.
+
+    :returns: updated resolved argument, ``str``
+    :raises: :exc:`ArgException` If arg invalidly specified
+    """
+    if len(args) == 0:
+        raise SubstitutionException(
+            '$(arg var) must specify a variable name [%s]' % (a))
+    elif len(args) > 1:
+        raise SubstitutionException(
+            '$(arg var) may only specify one arg [%s]' % (a))
+
+    if 'arg' not in context:
+        context['arg'] = {}
+    return resolved.replace('$(%s)' % a, _eval_arg(name=args[0], args=context['arg']))
+
+
+# Create a dictionary of global symbols that will be available in the eval
+# context.  We disable all the builtins, then add back True and False, and also
+# add true and false for convenience (because we accept those lower-case strings
+# as boolean values in XML).
+_eval_dict = {
+    'true': True, 'false': False,
+    'True': True, 'False': False,
+    '__builtins__': {k: __builtins__[k] for k in ['list', 'dict', 'map', 'str', 'float', 'int']},
+    'env': _eval_env,
+    'optenv': _eval_optenv,
+    'find': _eval_find
+}
+# also define all math symbols and functions
+_eval_dict.update(math.__dict__)
+
+
+def convert_value(value, type_):
+        """
```

Comment:
The indentation on this method is weird compared to the other ones in this file (and in this whole codebase).  Fix this to be 4 spaces like the rest.